src/choice.rs

```diff
@@ -0,0 +1,76 @@
+use core::convert::Infallible;
+
+use dusk_bytes::{HexDebug, Serializable};
+use subtle::ConditionallySelectable;
+
+#[cfg(feature = "canon")]
+use canonical_derive::Canon;
+
+#[cfg(feature = "rkyv-impl")]
+use bytecheck::CheckBytes;
+#[cfg(feature = "rkyv-impl")]
+use rkyv::{Archive, Deserialize as RkyvDeserialize, Serialize as RkyvSerialize};
+
+/// Wrapper for a [`subtle::Choice`]
+#[derive(Copy, Clone, HexDebug)]
+#[cfg_attr(feature = "canon", derive(Canon))]
+#[cfg_attr(feature = "rkyv-impl", derive(Archive, RkyvSerialize, RkyvDeserialize))]
+#[cfg_attr(feature = "rkyv-impl", archive_attr(derive(CheckBytes)))]
+pub struct Choice(u8);
```

Comment:
I have strong opinions against this struct that strives to achieve constant time computations, but then this PR is just extending what is already implemented. My suggestion is to remove `subtle` entirely, but its ofc not a blocking suggestion